app/workers/interval_worker.rb

```diff
@@ -8,12 +8,12 @@ def perform(id)
     @receiver = Receiver.find_by(id)
     return unless receiver
     receiver.event_actions.each(&:run!)
-    IntervalWorker.perform_at(start_time + (n * interval.seconds), id)
+    IntervalWorker.perform_at(receiver.start_time + (n * receiver.interval.seconds), id)
   end
 
   private
 
   def n
-    ((Time.zone.now.to_i - start_time) / interval).ceil
+    ((Time.zone.now.to_i - receiver.start_time) / receiver.interval).ceil
```

Comment:
Maybe make a method on the receiver `next_interval_time` to encapsulate this behaviour, so we can call that in the two spots we currently do this calculation?

Reply:
The reason I didn't do this initially is because I don't like making my concretions have different public interfaces than abstractions - becuase then you violate the Dependency Inversion principle when using this method of the public interface.

That being said, not really a big deal, and I see the benefit of having a shared method here.